styx-api-service/src/test/java/com/spotify/styx/api/WorkflowResourceTest.java

```diff
@@ -486,6 +504,103 @@ public void shouldPaginateWorkflowInstancesData() throws Exception {
     assertJson(response, "[0].workflow_instance.parameter", is("2016-08-12"));
   }
 
+  @Test
+  public void shouldReturnBadRequestWhenNoPayloadIsSentWorkflow() throws Exception {
+    sinceVersion(Api.Version.V3);
+
+    Response<ByteString> response =
+        awaitResponse(serviceHelper.request("POST", path("/foo")));
+
+    assertThat(response, hasStatus(withCode(Status.BAD_REQUEST)));
+    assertThat(response, hasNoPayload());
+    assertThat(response, hasStatus(withReasonPhrase(equalTo("Missing payload."))));
+  }
+
+  @Test
+  public void shouldReturnBadRequestWhenMalformedStatePayloadIsSentWorkflow() throws Exception {
+    sinceVersion(Api.Version.V3);
+
+    Response<ByteString> response =
+        awaitResponse(serviceHelper.request("POST", path("/foo"),
+            STATEPAYLOAD_BAD));
+
+    assertThat(response, hasStatus(withCode(Status.BAD_REQUEST)));
+    assertThat(response, hasNoPayload());
+    assertThat(response, hasStatus(withReasonPhrase(equalTo("Invalid payload."))));
+  }
+
+  @Test
+  public void shouldReturnOkWhenSchedulerReturnsSuccessWorkflow() throws Exception {
+    sinceVersion(Api.Version.V3);
+    serviceHelper.stubClient()
+        .respond(Response.forPayload(serialize(WORKFLOW2)))
+        .to(SCHEDULER_BASE + "/api/v0/workflows");
+
+    Response<ByteString> response =
+        awaitResponse(
+            serviceHelper.request("POST", path("/foo"), serialize(WORKFLOW_CONFIGURATION)));
+
+    assertThat(response, hasStatus(withCode(Status.OK)));
+    assertThat(deserialize(response.payload().get(), Workflow.class), equalTo(WORKFLOW2));
+  }
+
+  @Test
+  public void shouldReturnErrorMessageWhenSchedulerFailsWorkflow() throws Exception {
+    sinceVersion(Api.Version.V3);
+    serviceHelper.stubClient()
+        .respond(Response.forStatus(Status.SERVICE_UNAVAILABLE))
+        .to(SCHEDULER_BASE + "/api/v0/workflows");
+
+    Response<ByteString> response =
+        awaitResponse(
+            serviceHelper.request("POST", path("/foo"), serialize(WORKFLOW_CONFIGURATION)));
+
+    assertThat(response, hasStatus(withCode(Status.SERVICE_UNAVAILABLE)));
+    assertThat(response, hasNoPayload());
+  }
+
+  @Test
+  public void shouldReturnInternalServerErrorWhenSchedulerRequestFails() throws Exception {
+    sinceVersion(Api.Version.V3);
+
+    Response<ByteString> response =
+        awaitResponse(
+            serviceHelper.request("POST", path("/foo"), serialize(WORKFLOW_CONFIGURATION)));
```

Comment:
Added a comment to clarify this test. The stub client is intentionally not setup, this mimics a unreachable internal service (SchedulerResource). The path of the original call should still be /foo, because it corresponds to the component id (/<cid>) and should be a valid path in WorkflowResource  https://github.com/spotify/styx/pull/243/commits/08e7f13796a28c831f3e20f5b436cf9276d4614a